### PR TITLE
[api] fix web3 empty return

### PR DESCRIPTION
**Pull request by Liuhaai**
fix the bug that the server returns `null` instead of `[]` when calling web3 api `eth_getLogs`,

### CODE REVIEW

File: api/web3server.go [hunk 1]
```diff
@@ -831,31 +830,33 @@ func (svr *Web3Server) getFilterChanges(in interface{}) (interface{}, error) {
 	filterID = removeHexPrefix(filterID)
 	filterObj, err := loadFilterFromCache(svr.cache, filterID)
 	if err != nil {
-		return []interface{}{}, err
+		return nil, err
```
Comment (dustinxie) [inline api/web3server.go:833]: return nil is OK?
`nil` may cause panic if trying to json-decode it, but `[]interface{}{}` will not

File: api/web3server.go [hunk 2]
```diff
@@ -831,31 +830,33 @@ func (svr *Web3Server) getFilterChanges(in interface{}) (interface{}, error) {
...
-
 	switch filterObj.FilterType {
 	case "log":
+		if filterObj.LogHeight > tipHeight {
+			return []logsObject{}, nil
+		}
 		from, to, hasNewLogs, err := svr.getLogQueryRange(filterObj.FromBlock, filterObj.ToBlock, filterObj.LogHeight)
 		if err != nil {
 			return nil, err
 		} else if !hasNewLogs {
```
Comment (dustinxie) [inline api/web3server.go:848]: remove `else`

File: api/web3server_utils.go [hunk 3]
```diff
@@ -284,7 +284,7 @@ func (svr *Web3Server) getTransactionFromActionInfo(actInfo *iotexapi.ActionInfo
 
 func (svr *Web3Server) getTransactionCreateFromActionInfo(actInfo *iotexapi.ActionInfo) (transactionObject, error) {
 	tx, err := svr.getTransactionFromActionInfo(actInfo)
-	if err != nil {
+	if err != nil || tx == nil {
```
Comment (dustinxie) [inline api/web3server_utils.go:287]: can you check if there's other similar cases?

File: api/web3server.go (hunk 1 above)
Comment (Liuhaai) [reply api/web3server.go:833]: no. `func packAPIResult()` will only encode err

File: api/web3server_test.go [hunk 4]
```diff
@@ -642,3 +642,47 @@ func TestGetNetworkID(t *testing.T) {
...
 }
+
+func TestIsResultValid(t *testing.T) {
+	require := require.New(t)
+	testData := []struct {
+		data    web3Resp
+		isValid bool
+	}{
+		{
+			data: web3Resp{
```
Comment (dustinxie) [inline api/web3server_test.go:653]: can remove `data:` and `isValid:`

**Review by dustinxie — APPROVED** [review]

**Review by raullenchai — APPROVED** [review]

File: api/web3server.go [hunk 5]
```diff
@@ -623,7 +623,7 @@ func (svr *Web3Server) getTransactionReceipt(in interface{}) (interface{}, error
 		return nil, err
 	}
 	tx, err := svr.getTransactionFromActionInfo(actInfo)
-	if err != nil {
+	if err != nil || tx == nil {
```
Comment (raullenchai) [inline api/web3server.go:626]: why tx can be nil?

**Review by raullenchai — APPROVED** [review]

**Pull request merged**